vllm/model_executor/sampling_metadata.py

```diff
@@ -6,59 +6,284 @@
+                categorized_sample_indices_end_idx = (
+                    categorized_sample_indices_start_idx + subquery_len)
+                if do_sample:
+                    categorized_sample_indices_end_idx -= num_samples
+                prefill_indices = list(
+                    range(categorized_sample_indices_start_idx,
+                          categorized_sample_indices_end_idx))
+
+                # NOTE: prompt token positions do not need sample, skip
+                categorized_sample_indices_start_idx += subquery_len
+                if do_sample:
+                    categorized_sample_indices_start_idx -= num_samples
+
+            if do_sample:
+                # Update sample indices for this seq_group.
+                sample_indices = list(
+                    range(categorized_sample_indices_start_idx,
+                          categorized_sample_indices_start_idx + num_samples))
+                categorized_sample_indices[
+                    sampling_params.sampling_type].append(
+                        (categorized_sample_indices_start_idx,
+                         categorized_sampled_token_indices_start_idx))
+                categorized_sample_indices_start_idx += num_samples
+                categorized_sampled_token_indices_start_idx += num_samples
+
+            if sampling_params.seed is not None:
+                seq_group_metadata.state.generator = torch.Generator(
+                    device=device).manual_seed(sampling_params.seed)
+        else:
+            if do_sample:
+                num_seqs = len(seq_ids)
+                selected_token_indices.extend(
+                    range(selected_token_start_idx,
+                          selected_token_start_idx + num_seqs))
+                selected_token_start_idx += num_seqs
+
+                sample_indices = list(
+                    range(categorized_sample_indices_start_idx,
+                          categorized_sample_indices_start_idx + num_seqs))
+                categorized_sample_indices[
+                    sampling_params.sampling_type].extend(
+                        list(
+                            zip(
+                                range(
+                                    categorized_sample_indices_start_idx,
+                                    categorized_sample_indices_start_idx +
+                                    num_seqs),
+                                range(
+                                    categorized_sampled_token_indices_start_idx,
+                                    categorized_sampled_token_indices_start_idx
+                                    + num_seqs))))
+                categorized_sample_indices_start_idx += num_seqs
+                categorized_sampled_token_indices_start_idx += num_seqs
+
+        if sampling_params.seed is not None:
+            generator = seq_group_metadata.state.generator
+            # generators.append(seq_group_metadata.state.generator)
```

Comment:
Let's change this part to:
```python
if not do_sample:
    ...
else:
    if is_prompt:
        ...
    else:
        ...
```
I think in this case there will be less `if do_sample` in the code.